src/poprox_storage/repositories/articles.py

```diff
@@ -228,6 +241,15 @@ def store_articles(self, articles: list[Article], *, mentions=False, progress=Fa
 
         return failed
 
+    def store_candidates(self, articles: list[Article], date: date):
+        candidates_table = self.tables["candidate_articles"]
+        insert_stmt = (
+            insert(candidates_table)
+            .values([{"article_id": article.article_id, "date": date} for article in articles])
+            .on_conflict_do_nothing(constraint="uq_candidate_articles")
```

Comment:
Was just wondering if we could have a conflict in a "newsletter crashes and we manually re-start the workflow" situation, but I think this handles that.

Reply:
Candidate set ids would also help here (even though the uniqueness constraint mitigates the damage)